athena_abi/encode.go

```
@@ -0,0 +1,213 @@
+package athena_abi
+
+import (
+	"fmt"
+	"math/big"
+)
+
+func GetEnumIndex(enumType StarknetEnum, enumKey string) (int, StarknetType, error) {
+	for idx, variant := range enumType.Variants {
+		if variant.Name == enumKey {
+			return idx, variant.Type, nil
+		}
+	}
+	return 0, nil, &InvalidAbiError{Msg: fmt.Sprintf("enum key %s not found in enum %s", enumKey, enumType.Name)}
```

this is not an invalid abi error. you can just return an error string.